raiden_contracts/data_0.21.0/source/services/ServiceRegistry.sol

```diff
@@ -0,0 +1,324 @@
+        require(_decay_constant > 0, "attempt to set zero decay constant");
+        require(_decay_constant < 2 ** 40, "too big decay constant");
+        decay_constant = _decay_constant;
+    }
+
+    function set_registration_duration(uint256 _registration_duration) private {
+        // No checks.  Even allowing zero (when no new registrations are possible).
+        registration_duration = _registration_duration;
+    }
+
+
+    /// @notice The amount to deposit for registration or extension.
+    /// Note: the price moves quickly depending on what other addresses do.
+    /// The current price might change after you send a `deposit()` transaction
+    /// before the transaction is executed.
+    function current_price() public view returns (uint256) {
+        require(now >= set_price_at, "An underflow in price computation");
+        uint256 seconds_passed = now - set_price_at;
+
+        return decayed_price(set_price, seconds_passed);
+    }
+
+
+    /// @notice Calculates the decreased price after a number of seconds.
+    /// @param _set_price The initial price.
+    /// @param _seconds_passed The number of seconds passed since the initial
+    /// price was set.
+    function decayed_price(uint256 _set_price, uint256 _seconds_passed) public
+        view returns (uint256) {
+        // We are here trying to approximate some exponential decay.
+        // exp(- X / A) where
+        //   X is the number of seconds since the last price change
+        //   A is the decay constant (A = 200 days corresponds to 0.5% decrease per day)
+
+        // exp(- X / A) ~~ P / Q where
+        //   P = 24 A^4
+        //   Q = 24 A^4 + 24 A^3X + 12 A^2X^2 + 4 AX^3 + X^4
+        // Note: swap P and Q, and then think about the Taylor expansion.
+
+        uint256 X = _seconds_passed;
+
+        if (X >= 2 ** 40) { // The computation below overflows.
+            return min_price;
+        }
+
+        uint256 A = decay_constant;
+
+        uint256 P = 24 * (A ** 4);
+        uint256 Q = P + 24*(A**3)*X + 12*(A**2)*(X**2) + 4*A*(X**3) + X**4;
+
+        // The multiplication below is not supposed to overflow because
+        // _set_price should be at most 2 ** 90 and
+        // P should be at most 24 * (2 ** 40).
+        uint256 price = _set_price * P / Q;
+
+        // Not allowing a price smaller than min_price.
+        // Once it's too low it's too low forever.
+        if (price < min_price) {
+            price = min_price;
+        }
+        return price;
+    }
+}
+
+
+contract ServiceRegistry is Utils, ServiceRegistryConfigurableParameters {
+    Token public token;
+
+    mapping(address => uint256) public service_valid_till;
+    mapping(address => string) public urls;  // URLs of services for HTTP access
+
+    // An append-only list of addresses that have ever made a deposit.
+    // Starting from this list, all alive registrations can be figured out.
+    address[] public ever_made_deposits;
```

Comment:
So the only new change from the entire 43 files diff is this line?

Reply:
This is not new.  The Changelog entry is new, and the new version numbers are new.  Also `gas.json` numbers changed.

Reply:
I meant change compared to the previous version. I am getting this out of the changelog.